src/main/Core/FileSystemUtility/NodeJsFileSystemUtility.ts

```diff
@@ -60,7 +62,7 @@ export class NodeJsFileSystemUtility implements FileSystemUtility {
 
     public writeTextFile(data: string, filePath: string): Promise<void> {
         return new Promise((resolve, reject) => {
-            writeFile(filePath, data, "utf-8", (error) => {
+            writeFile(filePath, `${this.byteOrderMark}${data}`, "utf-8", (error) => {
```

Comment:
Can you explain what is happening here? 😅 Are you sure that this won't cause any issues on macOS and Linux?

Reply:
Powershell does not seem to recognize a file as UTF-8 if the [byte-order mark (BOM)](https://en.wikipedia.org/wiki/Byte_order_mark) is missing. Node does not seem to be able to write the BOM itself, so you have to add it manually. Source: https://stackoverflow.com/a/13859239/1836528

I did not test this on macOS or Linux since I don't have those OSes. I could set up a VM to test it on Linux but I don't have a Mac. Can you test it? Or we could just make it OS-dependent.

Reply:
No worries, I can test this on a Macbook next week. If this is an issue on macOS or Linux we can create OS-specific file system utility classes.